File: app/src/main/java/com/example/pokedex/PokemonActivity.java
package com.example.pokedex;

import android.os.Bundle;

import com.google.android.material.bottomnavigation.BottomNavigationView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.example.pokedex.databinding.ActivityPokemonBinding;

import dagger.hilt.android.AndroidEntryPoint;

@AndroidEntryPoint
public class PokemonActivity extends AppCompatActivity {

    private ActivityPokemonBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityPokemonBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        getSupportActionBar().hide();

        BottomNavigationView navView = findViewById(R.id.nav_view);
        // Passing each menu ID as a set of Ids because each
        // menu should be considered as top level destinations.
        AppBarConfiguration appBarConfiguration = new AppBarConfiguration.Builder(
                R.id.homeFragment, R.id.savedPokemon, R.id.searchPokemon)
                .build();
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_activity_pokemon);
        NavigationUI.setupActionBarWithNavController(this, navController, appBarConfiguration);
        NavigationUI.setupWithNavController(binding.navView, navController);
    }

}

File: app/src/main/java/com/example/pokedex/PokemonActivity.kt
package com.example.pokedex

import dagger.hilt.android.AndroidEntryPoint
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.example.pokedex.R
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.Navigation.findNavController
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration.Builder
import androidx.navigation.ui.NavigationUI
import androidx.navigation.ui.NavigationUI.setupActionBarWithNavController
import com.example.pokedex.R.id
import com.example.pokedex.databinding.ActivityPokemonBinding
import com.example.pokedex.utils.Helper.onNavDestinationSelected
import com.skydoves.whatif.whatIfNotNull

@AndroidEntryPoint
class PokemonActivity : AppCompatActivity() {
  private var _binding: ActivityPokemonBinding? = null
  private val binding get() = _binding!!
  private lateinit var navController: NavController

  override fun onCreateOptionsMenu(menu: Menu?): Boolean {
    menu.whatIfNotNull {
      menuInflater.inflate(R.menu.bottom_nav_menu, it)
      binding.navView.setupWithNavController(it, navController)
    }
    return true
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    installSplashScreen()
    super.onCreate(savedInstanceState)
    _binding = ActivityPokemonBinding.inflate(layoutInflater)
    setContentView(binding.root)
    supportActionBar!!.hide()
    navController = findNavController(id.nav_host_fragment_activity_pokemon)
  }
}